worldedit-core/src/main/java/com/sk89q/worldedit/world/registry/LegacyMapper.java

```diff
@@ -95,26 +95,34 @@ private void loadFromResource() throws IOException {
             String id = blockEntry.getKey();
             final String value = blockEntry.getValue();
             blockEntries.put(id, value);
-            try {
-                BlockState state = WorldEdit.getInstance().getBlockFactory().parseFromInput(value, parserContext).toImmutableState();
-                blockToStringMap.put(state, id);
-                stringToBlockMap.put(id, state);
-            } catch (InputParseException e) {
-                boolean fixed = false;
-                if (fixer != null) {
+
+            BlockState state = null;
+
+            // if fixer is available, try using that first, as some old blocks that were renamed share names with new blocks
+            if (fixer != null) {
+                try {
                     String newEntry = fixer.fixUp(DataFixer.FixTypes.BLOCK_STATE, value, 1631);
-                    try {
-                        BlockState state = WorldEdit.getInstance().getBlockFactory().parseFromInput(newEntry, parserContext).toImmutableState();
-                        blockToStringMap.put(state, id);
-                        stringToBlockMap.put(id, state);
-                        fixed = true;
-                    } catch (InputParseException ignored) {
-                    }
+                    state = WorldEdit.getInstance().getBlockFactory().parseFromInput(newEntry, parserContext).toImmutableState();
```

Comment:
Can you pull `WorldEdit.getInstance().getBlockFactory()` out to a variable? Seems appropriate.